test/source/tests/setup.ts

```diff
@@ -497,6 +437,28 @@ AN8G3r5Htj8olot+jm9mIa5XLXWzMNUZgg==
       await gmailPage.notPresent(['@webmail-notification-setup', '@notification-setup-action-open-settings', '@notification-setup-action-dismiss', '@notification-setup-action-close']);
     }));
 
+    ava.default('setup - test adding missing self-signature key issue', testWithBrowser('compatibility', async (t, browser) => {
+      const acctEmail = 'flowcrypt.compatibility@gmail.com';
+      const settingsPage = await browser.newPage(t, TestUrls.extensionSettings(acctEmail));
+      await SettingsPageRecipe.toggleScreen(settingsPage, 'additional');
+      const addKeyPopup = await SettingsPageRecipe.awaitNewPageFrame(settingsPage, '@action-open-add-key-page', ['add_key.htm']);
+      await addKeyPopup.waitAndClick('@source-paste');
+      const key = Config.key('missing.self.signatures');
+      await addKeyPopup.waitAndType('@input-armored-key', key?.armored ?? '');
+      await addKeyPopup.waitAndType('#input_passphrase', key?.passphrase ?? '');
+      await addKeyPopup.waitAndClick('.action_add_private_key', { delay: 1 });
+      await addKeyPopup.waitAll('@input-compatibility-fix-expire-years', { timeout: 30 });
+      await addKeyPopup.selectOption('@input-compatibility-fix-expire-years', '1');
+      await addKeyPopup.waitAndClick('@action-fix-and-import-key');
+      await Util.sleep(1);
+      const myKeyFrame = await SettingsPageRecipe.awaitNewPageFrame(settingsPage, `@action-show-key-2`, ['my_key.htm', 'placement=settings']);
+      await Util.sleep(1);
+      const curDate = new Date();
+      // getFullYear() + 1 because we set expiration as 1 year
+      const expiration = `${curDate.getFullYear() + 1}-${curDate.getMonth() + 1}-${curDate.getDate()}`;
+      expect(await myKeyFrame.read('@content-key-expiration')).to.equal(expiration);
```

Comment:
This test (or the key-fixing code) is susceptible to the time zone setting and/or date boundary.
![image](https://user-images.githubusercontent.com/23335944/202199847-af077b2b-c987-4e5d-8dee-78e1102d8b60.png)
![image](https://user-images.githubusercontent.com/23335944/202200257-7074cd28-2d0b-4547-b0cb-27ea58f8c98c.png)

=> AssertionError: expected '2023-11-16' to equal '2023-11-15'

Reply:
![image](https://user-images.githubusercontent.com/23335944/202200920-8ea862fd-6098-4b27-b459-eb39dbee8f4e.png)

Reply:
![image](https://user-images.githubusercontent.com/23335944/202201209-139f392b-a17a-4979-9016-a094d405b4c7.png)

Reply:
I honestly don't like current way of formatting date.
https://github.com/FlowCrypt/flowcrypt-browser/blob/128b3ea4c5207827225bd37e92ea1853cb3ea1aa/extension/chrome/settings/modules/my_key.ts#L58
![image](https://user-images.githubusercontent.com/77344191/202201693-d180f175-d80d-49f2-8ae4-9f577af92051.png)

Reply:
We may use `date-fns` (or other library like that) or make common `format` function which accepts `format` string.

Reply:
For now let me change above test code to below line (to let test pass despite of timezones)
But I suggest to implement date format functions like above comment and make new issue/PR for that.

```ts
const expirationDate = new Date(new Date().setFullYear(new Date().getFullYear() + 1));
const expiration = Str.datetimeToDate(Str.fromDate(expirationDate));
```

Reply:
What do you think @rrrooommmaaa ?

Reply:
> For now let me change above test code to below line (to let test pass despite of timezones)

It's certainly better, but may fail occasionally if the test is run at about 23:59 UTC
AssertionError: expected '2023-11-16' to equal '2023-11-17' (we can leave a leeway of minus one day) 

> But I suggest to implement date format functions like above comment and make new issue/PR for that.

For `dependencies` I'd ask for @tomholub 's approval.
For `devDependencies` we have more freedom )

Reply:
ok